### PR TITLE
Sol-37 connect gaps in data lines

### PR DESCRIPTION
**Pull request by billli0**
### Issue #:
https://linear.app/tupaia/issue/SOL-37/connecting-gaps-in-data-lines-and-change-x-axis-to-be-yearly

### Changes:
- add connectNulls in line chart to connect gaps

- add insertOrUpdateRootJsonEntry() in migration, because JSONB_SET can not modify root json entry.

- migration for config changes

### Screenshot 
See comment in card

### CODE REVIEW

File: packages/database/src/utilities/migration.js [hunk 1]
```diff
@@ -102,18 +102,39 @@ export async function updateValues(db, table, newValues, condition) {
   return db.runSql(`UPDATE "${table}" SET ${setQuery} WHERE ${whereQuery}`, params);
 }
 
+const getValueString = values => {
```
Comment (edmofro) [inline packages/database/src/utilities/migration.js:105]:
```suggestion
const getJsonColumnValueString = values => {
```

File: packages/database/src/utilities/migration.js [hunk 2]
```diff
@@ -102,18 +102,39 @@ export async function updateValues(db, table, newValues, condition) {
...
+      .map(item => (typeof item === 'string' ? `"${item}"` : item))
+      .join(',');
+    return `[${arrayString}]`;
+  }
+
+  if (typeof values === 'object') {
+    return JSON.stringify(values);
+  }
+
+  throw new Error('Function insertJsonEntry: we should either insert object or array');
```
Comment (edmofro) [inline packages/database/src/utilities/migration.js:117]:
```suggestion
  throw new Error('Function getJsonColumnValueString: we should either insert object or array');
```

**Review by edmofro — APPROVED** [review]
Nice one @billli0! Just one rename suggestion, which is up to you :-)

File: packages/database/src/utilities/migration.js (hunk 2 above)
Comment (billli0) [reply packages/database/src/utilities/migration.js:117]: Love it!

**Review by edmofro — APPROVED** [review]

**Pull request merged**